nixos/modules/services/web-servers/traefik.nix

```diff
@@ -0,0 +1,122 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.traefik;
+configFile =
+  if (cfg.configFile == null) then
+  (pkgs.runCommand "config.toml" {
+    buildInputs = [ pkgs.remarshal ];
+  } ''
+    remarshal -if json -of toml \
+    < ${pkgs.writeText "config.json" (builtins.toJSON cfg.configOptions)} \
+    > $out
+    '')
+    else
+    cfg.configFile;
+
+in {
+  options.services.traefik = {
+    enable = mkEnableOption "Traefik web server";
+
+    configFile = mkOption {
```

Comment:
What is your use case for this option then?

Reply:
I had forgotten to update the example to accurately portray the expected type: https://github.com/NixOS/nixpkgs/pull/29865/files#diff-beb91534bbf628b6e0727a0bb7d14c4aR25

Reply:
I mean in which case it is necessary to specify a toml path directly rather then using the `configOptions`?

Reply:
I would like to have it if I am doing an incremental change over to nixos from my current orchestration system for configuration files.

The other issue is [secrets](https://docs.traefik.io/configuration/acme/), which one may want to store in a separate store. This can be worked around for sure, but this gives people an alternative, less invasive route to initially integrating Traefik.

Are there any specific reasons why you think we _shouldn't_ have this option?
Reasons that come to mind:
- complexity
- against the ethos of NixOs
- allowing bad habits & configuration drift

For:
- already complex setups with templates & config-file generation.
- lower barrier for entry to users wanting to keep using traefik as-is with minimal overhead.

These issues aren't unsolvable, in fact nix is very good at solving them.
I'm happy to remove it if you think it won't find any use.

Reply:
I just want keep the API we expose to our users simples as we have to deal every option stable in future to not break configuration. But secrets is a good point for not storing the configuration outside of nix store: https://docs.traefik.io/configuration/backends/web/#basic-authentication

Reply:
What do you think then? We can keep it in pending #8 and deprecate it when secrets are properly managed within Nix

Reply:
Sorry, https://github.com/NixOS/nixpkgs/issues/24288. Wrong link

Reply:
it can stay then, I would say.